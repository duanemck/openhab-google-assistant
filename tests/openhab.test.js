const OpenHAB = require('../functions/openhab.js');

describe('Test EXECUTE', () => {
  test('OnOff Switch', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Switch"
        },
        "id": "MySwitch"
      }],
      "execution": [{
        "command": "action.devices.commands.OnOff",
        "params": {
          "on": false
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).not.toHaveBeenCalled();
    expect(sendCommandMock).toBeCalledWith('MySwitch', 'OFF');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MySwitch"
        ],
        "states": {
          "online": true,
          "on": false
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('mute Dimmer', async () => {
    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve());
    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Dimmer"
        },
        "id": "MySpeaker"
      }],
      "execution": [{
        "command": "action.devices.commands.mute",
        "params": {
          "mute": true
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(0);
    expect(sendCommandMock).toBeCalledWith('MySpeaker', '0');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MySpeaker"
        ],
        "states": {
          "isMuted": true,
          "online": true
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('setVolume Dimmer', async () => {
    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve());
    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Dimmer"
        },
        "id": "MySpeaker"
      }],
      "execution": [{
        "command": "action.devices.commands.setVolume",
        "params": {
          "volumeLevel": 40
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(0);
    expect(sendCommandMock).toBeCalledWith('MySpeaker', '40');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MySpeaker"
        ],
        "states": {
          "currentVolume": 40,
          "online": true
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('volumeRelative Dimmer', async () => {
    const item =
    {
      "state": "40",
      "type": "Dimmer",
      "name": "MySpeaker",
      "metadata": {
        "ga": {
          "value": "Speaker"
        }
      }
    };

    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));

    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Dimmer"
        },
        "id": "MySpeaker"
      }],
      "execution": [{
        "command": "action.devices.commands.volumeRelative",
        "params": {
          "relativeSteps": 20
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(sendCommandMock).toBeCalledWith('MySpeaker', '60');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MySpeaker"
        ],
        "states": {
          "currentVolume": 60,
          "online": true
        },
        "status": "SUCCESS"
      }]
    });
  });


  test('volumeRelative Dimmer max overflow', async () => {
    const item =
    {
      "state": "100",
      "type": "Dimmer",
      "name": "MySpeaker",
      "metadata": {
        "ga": {
          "value": "Speaker"
        }
      }
    };

    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));

    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Dimmer"
        },
        "id": "MySpeaker"
      }],
      "execution": [{
        "command": "action.devices.commands.volumeRelative",
        "params": {
          "relativeSteps": 20
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(sendCommandMock).toBeCalledWith('MySpeaker', '100');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MySpeaker"
        ],
        "states": {
          "currentVolume": 100,
          "online": true
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('volumeRelative Dimmer min overflow', async () => {
    const item =
    {
      "state": "10",
      "type": "Dimmer",
      "name": "MySpeaker",
      "metadata": {
        "ga": {
          "value": "Speaker"
        }
      }
    };

    const getItemMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));

    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Dimmer"
        },
        "id": "MySpeaker"
      }],
      "execution": [{
        "command": "action.devices.commands.volumeRelative",
        "params": {
          "relativeSteps": -20
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(sendCommandMock).toBeCalledWith('MySpeaker', '0');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MySpeaker"
        ],
        "states": {
          "currentVolume": 0,
          "online": true
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('OpenClose Rollershutter', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Rollershutter"
        },
        "id": "MyRollershutter"
      }],
      "execution": [{
        "command": "action.devices.commands.OpenClose",
        "params": {
          "openPercent": 0
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).not.toHaveBeenCalled();
    expect(sendCommandMock).toBeCalledWith('MyRollershutter', 'DOWN');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyRollershutter"
        ],
        "states": {
          "online": true,
          "openPercent": 0
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('OpenClose Rollershutter inverted', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Rollershutter",
          "inverted": true
        },
        "id": "MyRollershutter"
      }],
      "execution": [{
        "command": "action.devices.commands.OpenClose",
        "params": {
          "openPercent": 0
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).not.toHaveBeenCalled();
    expect(sendCommandMock).toBeCalledWith('MyRollershutter', 'UP');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyRollershutter"
        ],
        "states": {
          "online": true,
          "openPercent": 0
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('OpenClose Switch', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "customData": {
          "itemType": "Switch"
        },
        "id": "MyRollershutter"
      }],
      "execution": [{
        "command": "action.devices.commands.OpenClose",
        "params": {
          "openPercent": 0
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).not.toHaveBeenCalled();
    expect(sendCommandMock).toBeCalledWith('MyRollershutter', 'OFF');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyRollershutter"
        ],
        "states": {
          "online": true,
          "openPercent": 0
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('ColorAbsolute HSV', async () => {
    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "id": "MyColor"
      }],
      "execution": [{
        "command": "action.devices.commands.ColorAbsolute",
        "params": {
          "color": {
            "spectrumHSV": {
              "hue": 240.0,
              "saturation": 1.0,
              "value": 1.0
            }
          }
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).not.toHaveBeenCalled();
    expect(sendCommandMock).toBeCalledWith('MyColor', '240,100,100');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyColor"
        ],
        "states": {
          "online": true,
          "color": {
            "spectrumHsv": {
              "hue": 240.0,
              "saturation": 1.0,
              "value": 1.0
            }
          }
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('ColorAbsolute Temperature ColorLight', async () => {
    const item =
    {
      "state": "50,50,50",
      "type": "Color",
      "name": "MyColor",
      "metadata": {
        "ga": {
          "value": "LIGHT"
        }
      }
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "id": "MyColor"
      }],
      "execution": [{
        "command": "action.devices.commands.ColorAbsolute",
        "params": {
          "color": {
            "temperature": 4000
          }
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalled();
    expect(sendCommandMock).toBeCalledWith('MyColor', '26.97,34.9,50');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyColor"
        ],
        "states": {
          "online": true,
          "color": {
            "temperatureK": 4000
          }
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('ColorAbsolute Temperature SpecialColorLight', async () => {
    const item =
    {
      "type": "Group",
      "name": "MyColor",
      "metadata": {
        "ga": {
          "value": "LIGHT",
          "config": {
            "colorTemperatureRange": "1000,4000"
          }
        }
      },
      "members": [{
        "type": "Dimmer",
        "name": "MyBrightness",
        "metadata": {
          "ga": {
            "value": "lightBrightness"
          }
        },
        "state": "10"
      }, {
        "type": "Dimmer",
        "name": "MyTemperature",
        "metadata": {
          "ga": {
            "value": "lightColorTemperature"
          }
        },
        "state": "50"
      }]
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "id": "MyColor",
        "customData": {
          "deviceType": "SpecialColorLight"
        }
      }],
      "execution": [{
        "command": "action.devices.commands.ColorAbsolute",
        "params": {
          "color": {
            "temperature": 4000
          }
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalled();
    expect(sendCommandMock).toBeCalledWith('MyTemperature', '0');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyColor"
        ],
        "states": {
          "online": true,
          "color": {
            "temperatureK": 4000
          }
        },
        "status": "SUCCESS"
      }]
    });
  });

  test('OnOff SpecialColorLight', async () => {
    const item =
    {
      "type": "Group",
      "name": "MyColor",
      "metadata": {
        "ga": {
          "value": "LIGHT",
          "config": {
            "colorTemperatureRange": "1000,4000"
          }
        }
      },
      "members": [{
        "type": "Dimmer",
        "name": "MyBrightness",
        "metadata": {
          "ga": {
            "value": "lightBrightness"
          }
        },
        "state": "10"
      }, {
        "type": "Dimmer",
        "name": "MyTemperature",
        "metadata": {
          "ga": {
            "value": "lightColorTemperature"
          }
        },
        "state": "50"
      }]
    };

    const getItemMock = jest.fn();
    const sendCommandMock = jest.fn();
    getItemMock.mockReturnValue(Promise.resolve(item));
    sendCommandMock.mockReturnValue(Promise.resolve());

    const apiHandler = {
      getItem: getItemMock,
      sendCommand: sendCommandMock
    };

    const commands = [{
      "devices": [{
        "id": "MyColor",
        "customData": {
          "deviceType": "SpecialColorLight"
        }
      }],
      "execution": [{
        "command": "action.devices.commands.OnOff",
        "params": {
          "on": false
        }
      }]
    }];

    const payload = await new OpenHAB(apiHandler).handleExecute(commands);

    expect(getItemMock).toHaveBeenCalledTimes(1);
    expect(sendCommandMock).toBeCalledWith('MyBrightness', 'OFF');
    expect(payload).toStrictEqual({
      "commands": [{
        "ids": [
          "MyColor"
        ],
        "states": {
          "online": true,
          "on": false
        },
        "status": "SUCCESS"
      }]
    });
  });
});
