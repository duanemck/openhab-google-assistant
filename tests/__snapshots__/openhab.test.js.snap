// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SYNC Light Devices 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "SimpleLight",
        "itemType": "Switch",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Switch:MySwitch",
        "swVersion": "2.5.0",
      },
      "id": "MySwitch",
      "name": Object {
        "defaultNames": Array [
          "Light Switch",
        ],
        "name": "Light Switch",
        "nicknames": Array [
          "Light Switch",
          "Testlight",
          "Cool Light",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "DimmableLight",
        "itemType": "Dimmer",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Dimmer:MyDimmer",
        "swVersion": "2.5.0",
      },
      "id": "MyDimmer",
      "name": Object {
        "defaultNames": Array [
          "DimmLight",
        ],
        "name": "DimmLight",
        "nicknames": Array [
          "DimmLight",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {
        "colorModel": "hsv",
      },
      "customData": Object {
        "deviceType": "ColorLight",
        "itemType": "Color",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Color:MyLight",
        "swVersion": "2.5.0",
      },
      "id": "MyLight",
      "name": Object {
        "defaultNames": Array [
          "ColorLight",
        ],
        "name": "ColorLight",
        "nicknames": Array [
          "ColorLight",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
        "action.devices.traits.ColorSetting",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "SimpleLight",
        "itemType": "Switch",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Switch:MyLightGroup",
        "swVersion": "2.5.0",
      },
      "id": "MyLightGroup",
      "name": Object {
        "defaultNames": Array [
          "GroupLight",
        ],
        "name": "GroupLight",
        "nicknames": Array [
          "GroupLight",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "DimmableLight",
        "itemType": "Dimmer",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Dimmer:MyDimmerGroup",
        "swVersion": "2.5.0",
      },
      "id": "MyDimmerGroup",
      "name": Object {
        "defaultNames": Array [
          "GroupDimmer",
        ],
        "name": "GroupDimmer",
        "nicknames": Array [
          "GroupDimmer",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {
        "colorModel": "hsv",
      },
      "customData": Object {
        "deviceType": "ColorLight",
        "itemType": "Color",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Color:MyColorGroup",
        "swVersion": "2.5.0",
      },
      "id": "MyColorGroup",
      "name": Object {
        "defaultNames": Array [
          "GroupColor",
        ],
        "name": "GroupColor",
        "nicknames": Array [
          "GroupColor",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.Brightness",
        "action.devices.traits.ColorSetting",
      ],
      "type": "action.devices.types.LIGHT",
      "willReportState": false,
    },
    Object {
      "attributes": Object {},
      "customData": Object {
        "deviceType": "Fan",
        "itemType": "Dimmer",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Dimmer:MyFan",
        "swVersion": "2.5.0",
      },
      "id": "MyFan",
      "name": Object {
        "defaultNames": Array [
          "Fan",
        ],
        "name": "Fan",
        "nicknames": Array [
          "Fan",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.OnOff",
        "action.devices.traits.FanSpeed",
      ],
      "type": "action.devices.types.FAN",
      "willReportState": false,
    },
  ],
}
`;

exports[`Test SYNC Scene Device 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {
        "sceneReversible": true,
      },
      "customData": Object {
        "deviceType": "Scene",
        "itemType": "Switch",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Switch:MyScene",
        "swVersion": "2.5.0",
      },
      "id": "MyScene",
      "name": Object {
        "defaultNames": Array [
          "My Scene",
        ],
        "name": "My Scene",
        "nicknames": Array [
          "My Scene",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.Scene",
      ],
      "type": "action.devices.types.SCENE",
      "willReportState": false,
    },
  ],
}
`;

exports[`Test SYNC Sensor Device 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {
        "sensorStatesSupported": Object {
          "descriptiveCapabilities": Object {
            "availableStates": Array [
              "healthy",
              "moderate",
              "unhealthy",
              "very unhealthy",
            ],
          },
          "name": "AirQuality",
        },
      },
      "customData": Object {
        "deviceType": "Sensor",
        "itemType": "Number",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Number:MySensor",
        "swVersion": "2.5.0",
      },
      "id": "MySensor",
      "name": Object {
        "defaultNames": Array [
          "My Sensor",
        ],
        "name": "My Sensor",
        "nicknames": Array [
          "My Sensor",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.SensorState",
      ],
      "type": "action.devices.types.SENSOR",
      "willReportState": false,
    },
  ],
}
`;

exports[`Test SYNC Temperature Sensor Device 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {
        "queryOnlyTemperatureControl": true,
        "temperatureUnitForUX": "F",
      },
      "customData": Object {
        "deviceType": "TemperatureSensor",
        "itemType": "Number",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Number:MySensor",
        "swVersion": "2.5.0",
      },
      "id": "MySensor",
      "name": Object {
        "defaultNames": Array [
          "My Sensor",
        ],
        "name": "My Sensor",
        "nicknames": Array [
          "My Sensor",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.TemperatureControl",
      ],
      "type": "action.devices.types.SENSOR",
      "willReportState": false,
    },
  ],
}
`;

exports[`Test SYNC Thermostat Device 1`] = `
Object {
  "devices": Array [
    Object {
      "attributes": Object {
        "availableThermostatModes": "off,heat,eco,auto",
        "thermostatTemperatureUnit": "C",
      },
      "customData": Object {
        "deviceType": "Thermostat",
        "itemType": "Group",
      },
      "deviceInfo": Object {
        "hwVersion": "2.5.0",
        "manufacturer": "openHAB",
        "model": "Group:MyThermostat",
        "swVersion": "2.5.0",
      },
      "id": "MyThermostat",
      "name": Object {
        "defaultNames": Array [
          "My Thermostat",
        ],
        "name": "My Thermostat",
        "nicknames": Array [
          "My Thermostat",
        ],
      },
      "roomHint": undefined,
      "structureHint": undefined,
      "traits": Array [
        "action.devices.traits.TemperatureSetting",
      ],
      "type": "action.devices.types.THERMOSTAT",
      "willReportState": false,
    },
  ],
}
`;
